### PR TITLE
JS: promote the js/jwt-missing-verification query out of experimental

### PR DESCRIPTION
**Pull request by erik-krogh**
[Evaluation looks OK](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/CWE-347__default__CustomSuite__1/reports). No change in alerts, and performance that seems to be within the margin of error. 

The query flags basically the same issue as [this java query](https://github.com/github/codeql/blob/main/java/ql/src/Security/CWE/CWE-347/MissingJWTSignatureCheck.ql).

### CODE REVIEW

**Comment by kaeluka** [comment]
it looks good to me, but I'd like someone else to take a look as well :)

File: javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.ql [hunk 1]
```diff
@@ -10,12 +10,11 @@
...
 import javascript
-import DataFlow
 import semmle.javascript.RestrictedLocations
 
-from CallNode call
+from DataFlow::CallNode call
 where
-  call = moduleMember("jsonwebtoken", "verify").getACall() and
+  call = DataFlow::moduleMember("jsonwebtoken", "verify").getACall() and
   unique(boolean b | b = call.getArgument(1).analyze().getABooleanValue()) = false
```
Comment (esbena) [inline javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.ql:18]: This should be used instead: 
https://github.com/github/codeql/blob/ee52774e9079562ed30ce9998707f120de6192cb/javascript/ql/lib/semmle/javascript/dataflow/TypeInference.qll#L99-L100

File: javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.ql [hunk 2]
```diff
@@ -10,12 +10,11 @@
...
 select call.asExpr().(FirstLineOf),
```
Comment (esbena) [inline javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.ql:19]: Could we select the faulty argument instead with a message like:

`This argument disables the integrity enforcement of the token verification.`

**Review by esbena — DISMISSED** [review]

**Review by esbena — APPROVED** [review]

File: javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.qhelp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+<recommendation>
+<p>
+Calls to <code>verify()</code> functions should use a cryptographic secret or key to decode JWT payloads.
+</p>
+</recommendation>
+
+<example>
+<p>
+In the example below <code>false</code> is used to disable the integrity enforcement of a JWT payload.
```
Comment (mchammer01) [inline javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.qhelp:24]:
```suggestion
In the example below, <code>false</code> is used to disable the integrity enforcement of a JWT payload. 
```

File: javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.qhelp [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd"> <qhelp>
+
+<overview>
+<p>
+Applications decoding JSON Web Token (JWT) may be misconfigured due to the <code>None</code> algorithm.
```
Comment (mchammer01) [inline javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.qhelp:5]: Not sure about this suggestion, but token in its singular form reads oddly in this sentence.
```suggestion
Applications decoding JSON Web Tokens (JWT) may be misconfigured due to the <code>None</code> algorithm.
```

**Review by mchammer01 — APPROVED** [review]
@erik-krogh - LGTM ✨ 
A few minor suggestions and a typo.

File: javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.qhelp (hunk 4 above)
Comment (erik-krogh) [reply javascript/ql/src/Security/CWE-347/MissingJWTKeyVerification.qhelp:5]: The plural version seems is used often enough, and yes it sounds better :+1:

**Review by mchammer01 — APPROVED** [review]
Thanks for addressing my comments 💖

**Review by esbena — APPROVED** [review]

**Pull request merged**